pyresample/test/test_geometry.py

```diff
@@ -126,18 +141,87 @@ def test_create_areas_def(self):
                                             -909968.64000000001,
                                             1029087.28,
                                             1490031.3600000001])
-        import yaml
-        res = yaml.load(area_def.create_areas_def())
-        expected = yaml.load(('areaD:\n  description: Europe (3km, HRV, VTC)\n'
-                              '  projection:\n    a: 6378144.0\n    b: 6356759.0\n'
-                              '    lat_0: 50.0\n    lat_ts: 50.0\n    lon_0: 8.0\n'
-                              '    proj: stere\n  shape:\n    height: 800\n'
-                              '    width: 800\n  area_extent:\n'
-                              '    lower_left_xy: [-1370912.72, -909968.64]\n'
-                              '    upper_right_xy: [1029087.28, 1490031.36]\n'))
+        res = yaml.safe_load(area_def.create_areas_def())
+        expected = yaml.safe_load(('areaD:\n  description: Europe (3km, HRV, VTC)\n'
+                                   '  projection:\n    a: 6378144.0\n    b: 6356759.0\n'
+                                   '    lat_0: 50.0\n    lat_ts: 50.0\n    lon_0: 8.0\n'
+                                   '    proj: stere\n  shape:\n    height: 800\n'
+                                   '    width: 800\n  area_extent:\n'
+                                   '    lower_left_xy: [-1370912.72, -909968.64]\n'
+                                   '    upper_right_xy: [1029087.28, 1490031.36]\n'))
 
         self.assertDictEqual(res, expected)
 
+        # EPSG
+        area_def = geometry.AreaDefinition('baws300_sweref99tm', 'BAWS, 300m resolution, sweref99tm',
+                                           'sweref99tm',
+                                           'EPSG:3006' if utils.is_pyproj2() else {'init': 'epsg:3006'},
+                                           4667,
+                                           4667,
+                                           [-49739, 5954123, 1350361, 7354223])
+        res = yaml.safe_load(area_def.create_areas_def())
+        epsg_yaml = "EPSG: 3006" if utils.is_pyproj2() else 'init: epsg:3006'
```

Comment:
Don't we want to test both the new way and the old way? Regardless of pyproj version?

Reply:
Hmm, currently the new way `EPSG:XXXX` doesn't work with pyproj < 2. We would have to prepend `+init`. IIRC correctly we decided not to provide backwards compatibility.

Reply:
True, but this doesn't test the `init: epsg:3006` case with pyproj 2.0+. Is there a test that does do that?

Reply:
Ok, I see your point now. I forgot to test this combination. Will update the tests.